### DISCUSSION TITLE
Why are model component, water component and terrain component removable?

### DISCUSSION THREAD

**Discussion by Dgzt** [JamesTKhan/Mundus#181] [Q&A]
The model's model component, terrain's terrain component and water's water component are removable. But after remove them we can not add them again.  Why are these components removable?

**Comment by JamesTKhan** [comment]
I think its just because that is how ECS systems work. You normally don't restrict the ability to remove a component from a entity. Even if we did restrict removing them from the UI, it could still be removed programmatically. The only reason they can't be added back is that functionality has not been added in, and it might be  bit difficult (but not impossible) to allow adding a Terrain Component to be added to existing GO.
